### PR TITLE
add MT plugin for SAP Translation Hub

### PR DESCRIPTION
**Pull request by mlaggner**
A machine translation plugin for the SAP Translation Hub.
More infos at https://api.sap.com/shell/discover/contentpackage/SAPTranslationHub/api/translationhub

### CODE REVIEW

File: weblate/settings_example.py [hunk 1]
```diff
@@ -530,6 +531,13 @@
 # tmserver URL
 MT_TMSERVER = None
 
+# SAP Translation Hub
+MT_SAP_TRANSLATION_HUB_BASE_URL = None
+MT_SAP_TRANSLATION_HUB_SANDBOX_APIKEY = None
+MT_SAP_TRANSLATION_HUB_USERNAME = None
+MT_SAP_TRANSLATION_HUB_PASSWORD = None
+MT_SAP_TRANSLATION_HUB_USE_MT = True
```
Comment (nijel) [inline weblate/settings_example.py]: These should be added to `weblate/trans/models/conf.py` in order to have default value if user doesn't specify them in the `settings.py`.

Comment (mlaggner) [reply weblate/settings_example.py]: added this as you requested

File: weblate/trans/models/conf.py [hunk 2]
```diff
@@ -68,6 +68,13 @@ class WeblateConf(AppConf):
     # tmserver URL
     MT_TMSERVER = None
 
+    # SAP Translation Hub
+    MT_SAP_TRANSLATION_HUB_BASE_URL = None
+    MT_SAP_TRANSLATION_HUB_SANDBOX_APIKEY = None
```
[the remark this answers is not in the document]
Comment (mlaggner) [reply weblate/trans/models/conf.py]: is that name too long or where is the problem?

[the remark this answers is not in the document]
Comment (nijel) [reply weblate/trans/models/conf.py]: Yes, that's probably on what it's complaining.

**Comment by nijel** [comment]
Thanks, I've merged your changes. I've decided to use a but shorter name for the variables, see 0976ff1.

**Pull request merged**